src/philipstv.ts

```diff
@@ -250,28 +250,50 @@ export class PhilipsTV {
         const url = 'https://' + this.ip + ':1926/' + String(this.config.apiVersion) + '/audio/volume';
         const request_body = { 'muted': muted, 'current': this.volume };
         const result = await post(url, JSON.stringify(request_body), this.auth!);
-        return JSON.parse(result);    
+        return JSON.parse(result);
     }
 
     async sendKey(key: string) {
         const url = 'https://' + this.ip + ':1926/' + String(this.config.apiVersion) + '/input/key';
         const request_body = { 'key': key };
         const result = await post(url, JSON.stringify(request_body), this.auth!);
-        return JSON.parse(result);   
+        return JSON.parse(result);
     }
 
     async launchApplication(application: Record<string, string>) {
         const url = 'https://' + this.ip + ':1926/' + String(this.config.apiVersion) + '/activities/launch';
         const result = await post(url, JSON.stringify(application), this.auth!);
-        return JSON.parse(result);   
+        return JSON.parse(result);
     }
 
     async launchTVChannel(application: Record<string, string>) {
         const url = 'https://' + this.ip + ':1926/' + String(this.config.apiVersion) + '/activities/tv';
         const result = await post(url, JSON.stringify(application), this.auth!);
-        return JSON.parse(result); 
+        return JSON.parse(result);
+    }
+
+    async getAmbilightState() {
+        const url = 'https://' + this.ip + ':1926/' + String(this.config.apiVersion) + '/ambilight/power';
+        const result = await get(url, '', this.auth!);
+        const ambilightState = JSON.parse(result);
+        return ambilightState.power === 'On';
     }
 
+    async setAmbilightState(state: boolean) {
+        if (state) {
+            const url = 'https://' + this.ip + ':1926/' + String(this.config.apiVersion) + '/ambilight/currentconfiguration';
+            return await post(url, JSON.stringify({
```

Comment:
Why return await here, if there is no try catch around it will have no effect, just return the post it needs to be awaited from calling method anyway.